### PR TITLE
Deconstructing walls without repairing them

### PR DESCRIPTION
**Pull request by HumiliatedGoblin**
# About the pull request

You now need to have disarm or above to slice through walls, and help intent to repair walls.

# Explain why it's good for the game

Comtechs now don't need to spend 30 seconds repairing a wall, waste like 5 units of welding fuel only to deconstruct it. Now you can have choice in what you do! :D

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

it works trust

</details>

# Changelog

:cl:
qol: Now needs help intent to repair walls
qol: Now needs disarm, grab and harm intent to deconstruct walls
/:cl:

### CODE REVIEW

File: code/game/turfs/walls/walls.dm [hunk 1]
```diff
@@ -482,6 +482,8 @@
 /turf/closed/wall/proc/try_weldingtool_usage(obj/item/W, mob/user)
 	if(!damage || !iswelder(W))
 		return FALSE
+	if(!(user.a_intent == INTENT_HELP))
```
Comment (ihatethisengine) [inline code/game/turfs/walls/walls.dm:485]:
```suggestion
	if(user.a_intent != INTENT_HELP)
```

**Review by harryob — COMMENTED** [review]
we might need some text adding to the examine strings to make this more obvious beyond a changelog entry - thoughts?

**Comment by HumiliatedGoblin** [comment]
> we might need some text adding to the examine strings to make this more obvious beyond a changelog entry - thoughts?

Do you think it would be better to have info on the wall, or on the blowtorch? Or both?
My idea is to just put it underneath the weld_state string, but maybe you'd have a better idea

**Comment by HumiliatedGoblin** [comment]
@harryob  HARRY PLEASE

**Review by Drulikar — APPROVED** [review]

**Pull request merged**